app/app/services/sftp_case_transmitter.rb

```diff
@@ -0,0 +1,34 @@
+class SftpCaseTransmitter
+  attr_reader :current_agency, :cbv_flow, :aggregator_report
+
+  def initialize(cbv_flow, current_agency, aggregator_report)
+    @cbv_flow = cbv_flow
+    @current_agency = current_agency
+    @aggregator_report = aggregator_report
+  end
+
+  def deliver_sftp!
+    config = current_agency.transmission_method_configuration
+    sftp_gateway = SftpGateway.new(config)
+    transmitted_time = Time.now
+    filename = ClientAgency::AzDes::Configuration.pdf_filename(cbv_flow, transmitted_time)
+    sftp_gateway.upload_data(pdf_output.content, "#{config["sftp_directory"]}/#{filename}.pdf")
```

Comment:
`Net::SFTP::Session#upload!` uploads an IO subclass. As such:

```suggestion
    sftp_gateway.upload_data(StringIO.new(pdf_output.content), "#{config["sftp_directory"]}/#{filename}.pdf")
```